packages/suite/src/middlewares/wallet/fiatRatesMiddleware.ts

```diff
@@ -12,73 +13,70 @@ const fiatRatesMiddleware =
+            const prevAccount = prevState.wallet.accounts.find(
+                a => a.descriptor === account.descriptor && a.symbol === account.symbol,
+            );
 
-                if (account.tokens) {
-                    const difference = account.tokens.filter(
-                        token =>
-                            !prevAccount?.tokens?.find(
-                                prevToken => prevToken.symbol === token.symbol,
-                            ),
-                    );
+            if (account.tokens) {
+                const difference = account.tokens.filter(
+                    token =>
+                        !prevAccount?.tokens?.find(prevToken => prevToken.symbol === token.symbol),
+                );
 
-                    difference.forEach(token => {
-                        if (token.symbol) {
-                            api.dispatch(
-                                fiatRatesActions.updateCurrentRates({
-                                    symbol: token.symbol,
-                                    mainNetworkSymbol: account.symbol,
-                                    tokenAddress: token.address,
-                                }),
-                            );
-                        }
-                    });
-                }
-                break;
-            }
-            case ACCOUNT.CREATE: {
-                // fetch current rates for account's tokens
-                const account = action.payload;
-                account.tokens?.forEach(token => {
-                    if (!token.symbol) {
-                        return;
+                difference.forEach(token => {
+                    if (token.symbol) {
+                        api.dispatch(
+                            fiatRatesActions.updateCurrentRates({
+                                symbol: token.symbol,
+                                mainNetworkSymbol: account.symbol,
+                                tokenAddress: token.address,
+                            }),
+                        );
                     }
-                    api.dispatch(
-                        fiatRatesActions.updateCurrentRates({
-                            symbol: token.symbol,
-                            mainNetworkSymbol: account.symbol,
-                            tokenAddress: token.address,
-                        }),
-                    );
                 });
-
-                break;
             }
-            case WALLET_SETTINGS.CHANGE_NETWORKS:
-                api.dispatch(fiatRatesActions.removeRatesForDisabledNetworks());
-                break;
-            case WALLET_SETTINGS.SET_LOCAL_CURRENCY:
-                // for coins relying on coingecko we only fetch rates for one fiat currency
-                api.dispatch(fiatRatesActions.updateLastWeekRates());
-                break;
-            default:
-                break;
+        } else if (accountsActions.createAccount.match(action)) {
```

Comment:
there is no reason for else if, use just two `if` a do not place `switch` into `else`. It will work just same but will be cleaner.